### ISSUE TITLE
Clarify what the device version property should report

### ISSUE THREAD

**Issue by npmiller** [KhronosGroup/SYCL-Docs#222]
The device property `info::device::version` is defined as:

> Returns the SYCL version as a std::string in the form: <major_version>.<minor_version>.

In section [4.6.4.2](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_device_information_descriptors), Table 25.

But the current SYCL version is `2020` so what should this property return?

In addition this property is also mentioned in the OpenCL backend spec with the following, confusing paragraph:

> The device version is an indication of the device’s capabilities, as represented by the device information returned by the sycl::device::get_info() member function. Examples of attributes associated with the device version are resource limits and information about functionality beyond the requirements in the [core SYCL specification](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#core-spec). The version returned corresponds to the highest version of the OpenCL specification for which the device is conformant, but is not higher than the version of the device’s platform which bounds the overall capabilities of the runtime operating the device.

In section [C.5.1](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_platform_mixed_version_support_2).

This paragraph seems to suggest that the device version informs on hardware capabilities, but then turns around to say it only represent the highest OpenCL version supported.

And this also conflicts with the definition in section [4.6.4.2](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_device_information_descriptors) that says it should be the SYCL version.

In summary it would be good to clarify:
* What is the meaning of value reported by `info::device::version`?
* What is the format of that value?
* Can a backend change the meaning of that value and report something different?

**Comment by fraggamuffin** [comment]
definition is now year based +revision
Device or device driver version? Should platform version be also used?
use along with backend info descriptor query
Nicolas to draft based on today's discussion
not SYCL version anymore, but platform driver version is BE dependent

**Comment by keryell** [comment]
I think this versioning was following the OpenCL model but this has no longer a meaning in the general case of SYCL 2020 with various back-ends.
